File: data/multilang_comparison/program_julia/src/func23.c
#include "program_julia.h" 
array_t* func23(array_t_param* vars, int loopsFactor) {
   size_t pCounter = vars->size;
   array_t* array66;
   if (pCounter > 0) {
      array66 = vars->data[--pCounter];
      array66->refC++;
      DEBUG_COPY(array66->id);
   } else {
      array66 = (array_t*)malloc(sizeof(array_t));
      array66->size = 139;
      array66->refC = 1;
      array66->id = 66;
      array66->data = (unsigned int*)malloc(array66->size*sizeof(unsigned int));
      memset(array66->data, 0, array66->size*sizeof(unsigned int));
      DEBUG_NEW(array66->id);
   }
   array_t_param params0;
   params0.size = 1;
   params0.data = (array_t**)malloc(params0.size*sizeof(array_t*));
   params0.data[0] = array66;
   array_t* array67 = func28(&params0, get_path(), loopsFactor);
   DEBUG_RETURN(array67->id);
   free(params0.data);
   unsigned int loop21 = 0;
   unsigned int loopLimit21 = (rand()%loopsFactor)/4 + 1;
   for(; loop21 < loopLimit21; loop21++) {
      for (int i = 0; i < array67->size; i++) {
         array67->data[i]++;
      }
   }
   for (int i = 0; i < array66->size; i++) {
      if (array66->data[i] == 42) { 
         return array66;
      }
   }
   array67->refC--;
   if(array67->refC == 0) {
      free(array67->data);
      free(array67);
      DEBUG_FREE(array67->id);
   }
   return array66;
}

File: data/multilang_comparison/program_julia/src/func23.jl
include("program_julia_head.jl")
function func23(vars::JArray_param, loopsFactor::Int)::JArray
   pCounter = length(vars.data)
   array45 = nothing
   if pCounter > 1
       array45 = vars.data[pCounter]
       pCounter -= 1
       array45.refC += 1
   else
       array45 = JArray(zeros(Int, 555), 555, 1, 45)
   end
   params0 = JArray_param{Int}()
   params0.size = 1
   push!(params0.data, array45)
   params0.size += 1
   array46 = func28(params0, get_path(), loopsFactor);
   loopLimit15 = (50)/4 + 1
   for i in 1:loopLimit15 
      for i in 0:(array45.size-1) 
         array45.data[i+1] += 1
      end
   end
   for i in 1:array45.size
       if array45.data[i] == 41
               return array45
      end
   end
   array46.refC -= 1
   if array46.refC == 0
       empty!(array46.data)
       array46 = nothing
   end
   return array45;
end

